File: 2089-find-target-indices-after-sorting-array/2089-find-target-indices-after-sorting-array.java
class Solution {
    public List<Integer> targetIndices(int[] nums, int target) {
        List<Integer> array = new ArrayList<>();
        for (int num : nums) {
            while (true) {
                if (array.size() > num-1) {
                    break;
                }
                array.add(0);
            }
            array.set(num-1, array.get(num-1)+1);
        }

        int count = 0;
        List<Integer> answer = new ArrayList<>();
        for (int i = 0; i < array.size(); i++) {
            if (target -1 == i) {
                for (int j = 0; j < array.get(i); j++) {
                    answer.add(count+j);
                }
                break;
            }
            count += array.get(i);
        }
        return answer;
    }
}

File: 2089-find-target-indices-after-sorting-array/2089-find-target-indices-after-sorting-array.go
func targetIndices(nums []int, target int) []int {
	array := make([]int, 0)
	for _, num := range nums {
		for {
			if len(array) > num-1 {
				break
			}
			array = append(array, 0)
		}
		array[num-1]++
	}
	
	count := 0
	ans := make([]int, 0)
	for i := range array {
		if i == target -1 {
			for j := 0; j < array[i]; j++ {
				ans = append(ans, count + j)
			}
			break
		}
		count += array[i]
	}
	return ans
}